src/Node.hs

```diff
@@ -281,11 +282,12 @@ node
        )
     => NT.Transport m
     -> StdGen
+    -> RL.RateLimiting m
     -> packing
     -> peerData
     -> (Node m -> NodeAction packing peerData m t)
     -> m t
-node transport prng packing peerData k = do
+node transport prng rateLimiting packing peerData k = do
```

Comment:
The `RateLimiting m` contains a `forall t . IO (QDisc t)`, but the `Transport m` presumably has one of these as well. I suppose it's the programmer's responsibility to ensure that the transport is created using `rlQDisc`. I think we should make this more clear by having the `RateLimiting m` parameter come first, and then take a `(forall t. IO (QDisc t)) -> NT.Transport m` which will always be called with the `rlQDisc` of the `RateLimiting m`.